Widget/Fitness/FitnessWidget.swift

```diff
@@ -0,0 +1,237 @@
+
+    if(date < hours.0) {
+        return "Closed • Opens at " + dateFormatter.string(from: hours.0).replacingOccurrences(of: ":00", with: "")
+    } else if (hours.0 <= date && date <= hours.1) {
+        return "Open until " + dateFormatter.string(from: hours.1).replacingOccurrences(of: ":00", with: "") + ((showbusystring) ? (" • " + getBusyString(room: room)) : "")
+    } else {
+        return "Closed"
+    }
+}
+
+private func getHours (room: FitnessRoom) -> (Date, Date) {
+    let weekdayIndex: Int = (Calendar.current.component(.weekday, from: Date()) + 5) % 7
+    var hours: (Date, Date) {
+        let calendar = Calendar.current
+        let currentDate = Date()
+        let timeFormatter = DateFormatter()
+        timeFormatter.dateFormat = "HH:mm:ss"
+        let openTime = timeFormatter.date(from: room.open[weekdayIndex])!
+        let closeTime = timeFormatter.date(from: room.close[weekdayIndex])!
+
+        let openDate = calendar.date(bySettingHour: openTime.hour, minute: openTime.minutes, second: 0, of: currentDate)!
+        let closeDate = calendar.date(bySettingHour: closeTime.hour, minute: closeTime.minutes, second: 0, of: currentDate)!
+        return (openDate, closeDate)
+    }
+    return hours
+}
+
+private func isOpen (room: FitnessRoom) -> Bool {
+    let date = Date()
+    let hours = getHours(room: room)
+    return hours.0 <= date && date <= hours.1
+}
+
+
+
+private struct FitnessWidgetSmallView: View {
+    var room: FitnessRoom
+    var configuration: ConfigureFitnessWidgetIntent.Configuration
+    
+    
+    var body: some View {
+        VStack {
+            Text(room.name)
+                .font(.system(size: 13, weight: .semibold))
+                .padding(.top)
+                .foregroundStyle(configuration.background.prefersGrayscaleContent  ? Color.primary : .black)
+            Text(getOpenString(room: room, showbusystring: false))
+                .font(.system(size: 11))
+                .foregroundStyle(configuration.background.prefersGrayscaleContent ? Color.primary : (isOpen(room: room) ? .green : .blue))
+                .multilineTextAlignment(.center)
+                .fixedSize(horizontal: false, vertical: true)
+            MeterView(current: isOpen(room: room) ? room.capacity : 0, maximum: 100.0, style: (configuration.background.prefersGrayscaleContent  ? Color.primary : .blue), lineWidth: 6) {
+                VStack {
+                    Text("\(isOpen(room: room) ? room.capacity : 0, specifier: "%.2f")%")
+                        .fontWeight(.bold)
+                        .foregroundStyle(configuration.background.prefersGrayscaleContent  ? Color.primary : .black)
+                    Text("capacity")
+                        .font(.system(size: (configuration.background.prefersGrayscaleContent ? 13 : 10), weight: .light, design: .default))
+                        .foregroundStyle(configuration.background.prefersGrayscaleContent  ? Color.primary : .black)
```

Comment:
I'm not sure if this should always be black?

Reply:
Yeah, just leave the two at `Color.primary` - it's already grayscale, so no need to use `.black`